lms/product/canvas/_plugin/grouping.py

```diff
@@ -19,15 +19,23 @@ class CanvasGroupingPlugin(GroupingPlugin):
     group_type = Grouping.Type.CANVAS_GROUP
     sections_type = Grouping.Type.CANVAS_SECTION
 
-    def __init__(self, canvas_api):
+    def __init__(self, canvas_api, strict_section_membership: bool):
         self._canvas_api = canvas_api
+        self._strict_section_membership = strict_section_membership
 
     def get_sections_for_learner(self, _svc, course):
         return self._canvas_api.authenticated_users_sections(
             self._custom_course_id(course)
         )
 
-    def get_sections_for_instructor(self, _svc, course):
+    def get_sections_for_instructor(self, svc, course):
+        if self._strict_section_membership:
+            # Return only sections the current instructor belongs to
+            return self._canvas_api.authenticated_users_sections(
```

Comment:
This is the same behavior as for learners, see the `get_sections_for_learner` method above